icicle/tests/test_curve_api.cpp

```diff
@@ -319,6 +300,26 @@ TYPED_TEST(CurveSanity, CurveSanityTest)
   ASSERT_EQ(a - b, a - TypeParam::to_affine(b)); // mixed subtraction projective-affine
 }
 
+TYPED_TEST(CurveSanity, ScalarMultTest)
+{
+  const auto point = TypeParam::rand_host();
+  const auto scalar = scalar_t::rand_host();
+
+  START_TIMER(main)
+  const auto mult = scalar * point;
+  END_TIMER(main, "scalar mult window method", true);
+
+  auto expected_mult = TypeParam::zero();
+  START_TIMER(ref)
+  for (int i = 0; i < scalar_t::NBITS; i++) {
+    if (i > 0) { expected_mult = TypeParam::dbl(expected_mult); }
+    if (scalar.get_scalar_digit(scalar_t::NBITS - i - 1, 1)) { expected_mult = expected_mult + point; }
+  }
+  END_TIMER(ref, "scalar mult double-and-add", true);
+
+  ASSERT_EQ(mult, expected_mult);
```

Comment:
Is there a comparison to be made between CPU and GPU like the other tests?

Reply:
no it's just a test for me to be sure I did not mess the logic. Because both CUDA and CPU use the same source code for EC*scalar so they will agree even if the computation is wrong.

Then it was useful to compare to the old method perfromance.